### PR TITLE
3.2 into 3.3

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Forward merge:
- https://github.com/juju/juju/pull/16034
- https://github.com/juju/juju/pull/16042
- https://github.com/juju/juju/pull/16044
- https://github.com/juju/juju/pull/16047
- https://github.com/juju/juju/pull/16048

One trivial conflict with imports in `core/charm/repository/charmhub.go`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -378,6 +378,9 @@ go-agent-build-no-cgo: $(BUILD_AGENT_TARGETS)
 .PHONY: go-client-build
 go-client-build: $(BUILD_CLIENT_TARGETS)
 
+.PHONY: go-client-build
+go-client-build: $(BUILD_CLIENT_TARGETS)
+
```
Comment (barrettj12) [inline Makefile:383]: this is duplicated, remove

**Review by barrettj12 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (barrettj12) [reply Makefile:383]: Actually, looks like it got duplicated in 3.2 as well, so I'll remove it there. Leave this to avoid conflicts

Comment (jack-w-shaw) [reply Makefile:383]: Oops

**Pull request merged**